src/TestFramework/TestFramework/Attributes/TestMethod/ConditionBaseAttribute.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestTools.UnitTesting;
+
+public enum Mode
+{
+    Include,
+    Exclude,
+}
+
+/// <summary>
+/// This attribute is used to conditionally control whether a test class or a test method will run or be ignored, based on a condition and using an optional message.
+/// </summary>
+/// <remarks>
+/// This attribute isn't inherited. Applying it to a base class will not affect derived classes.
+/// </remarks>
+[AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, Inherited = false)]
+public abstract class ConditionBaseAttribute : Attribute
+{
+    protected ConditionBaseAttribute(Mode mode)
+        => Mode = mode;
+
+    protected ConditionBaseAttribute()
```

Comment:
Do we want to allow this one or force subclasses to select the mode?

Reply:
We can force for now. Easy it to introduce the ctor later **if** we wantaed.